dals/fileFunctions.js

```diff
@@ -1,36 +1,66 @@
 const mongoose = require('mongoose');
-const Issue = mongoose.model('Issue');
-const File = mongoose.model('File');
-
-exports.saveFile = (req, res) => {
-    const data = req.payload;
-    if(data.file){
-        File.save({
-            url: data.file.hapi.filename,
-            issue: issue._id
-        }, (err, issue) => {
-            if(err){
-                reply(err).code(500);
-            }
-            return res.response(issue);
+const fs = require('fs');
+const mime = require('mime-types');
+
+const path = require('path');
+const Issue = require(path.resolve('./models/issueModel'));
+const File = require('../models/fileModel');
+
+const internals = {};
+
+internals.saveFile = (file, issueId, resolve, fileName) => {
```

Comment:
Whenever there are more than two arguments in the function you should put the in one, 'params' argument. Something like this:

```
internals.saveFile = (params) => {
    const { file, issueId, resolve, fileName } = params;
    // ili
    const file = params.file;
    const issueId = params.issueId;
    ...
}
```
Also, this function should accept callback - function that will be called once issue is found (line #24). Something like this:

```
internals.saveFile = (params, callback) => {
    const { file, issueId, resolve, fileName } = params;
    Issue.findOne({ _id: issueId }, (err, issue) => {
        callback(err, issue);
    })
}
```